src/components/PopupWithForm.js

```diff
@@ -1,36 +1,46 @@
-import PopUp from "./PopUp.js";
+import PopUp from "./Popup.js";
 
 export default class PopUpWithForm extends PopUp {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputEls = Array.from(
-      this._popupForm.querySelectorAll(".modal__input")
-    );
+      this._popupForm.querySelectorAll(".modal__input"));
+    this._popupButton = this._popupForm.querySelector(".modal__button");
   }
+  
 
   close() {
     this._popupForm.reset();
     super.close();
   }
 
   _getInputValues() {
-    const formValues = [];
+    const formValues = {};
     this._inputEls.forEach((element) => {
       formValues[element.name] = element.value;
     });
 
     return formValues;
   }
 
+  submitLoading(isLoading) {
```

Comment:
`submitLoading`

Function names should start with a **verb** in infinitive (there is such a point in the check list). It’s common in coding to do it so that functions could `speak` for themselves about what they do.


`setButtonText` or `renderLoading`